src/inmanta/config.py

```diff
@@ -180,7 +180,7 @@ def validate_option_request(cls, section: str, name: str, default_value: Optiona
             # raise Exception("Config section %s not defined" % (section))
             return None
         if name not in cls.__config_definition[section]:
-            LOGGER.warning("Config name %s not defined in section %s" % (name, section))
+            LOGGER.warning(f"Config name {name} not defined in section {section}")
```

Comment:
I think I recall there were some security considerations to prefer passing the args to the logger to let it do the interpolation but I can only find performance considerations online. Do you know about it ?

Reply:
In Python the security aspect is pretty limited, but one possibility is when we don't fully control the interpolations, they can have "%s" in them. In that case the logger will expect a corresponding parameter but there isn't one, so it will trigger an exception. Afaik that's about the worst that can happen in Python (C is **a lot** worse), but still good practice to just  use the loggers as intended.

```
import logging


name = "injecting my own %s interpolation"
# This one just prints a literal "%s" because there are no logger args. It's entirely possible that this in a safeguard for exactly this misusage
logging.warning(f"hello {name}")
# As soon as there's one logger arg we run in trouble: logger expects 2 but there's only 1
logging.warning(f"hello %s and {name}", "world")
```